src/Form/Checkbox.js

```diff
@@ -111,47 +115,80 @@ const CheckboxLabel = createComponent({
     const sizeStyles = getComponentSize(theme, 'CheckboxLabel', size);
 
     return css`
-      margin-left: 8px;
+      transition: color ${transitionTiming};
 
       ${sizeStyles}
     `;
   },
 });
 
 const CheckboxContainer = createComponent({
-  name: 'Checkbox',
+  name: 'CheckboxContainer',
   tag: 'label',
-  style: ({ horizontal, checked, theme, disabled }) => css`
-    position: relative;
-    margin-bottom: 0;
-    cursor: pointer;
-    display: flex;
-    align-items: flex-start;
+  style: ({ theme, isChecked, isDisabled, isHorizontal, size, color }) => {
+    const sizeStyles = getComponentSize(theme, 'CheckboxContainer', size);
 
-    & + & {
-      margin-left: ${horizontal ? 12 : 0}px;
-      margin-top: ${horizontal ? 0 : 4}px;
-    }
+    return css`
+    display: inline-flex;
+    align-items: flex-start;
+    justify-content: flex-start;
+    margin-bottom: 4px;
+    user-select: none;
+    cursor: pointer;
```

Comment:
stylistic nit: can we indent these styles here and below 2 spaces to the right to be consistent with other styles? 

```js
return css`
  display: inline-flex;
  // etc 
`
```